### PR TITLE
(update) Migrating website-v9 header to lens

### PR DESCRIPTION
**Pull request by carlagn**
## Describe this PR

Website-v9 has the most recent version of the header, implemented using `@emotion/styled` and `@emotion/react` for the styled components. The goal of this PR is to migrate these components from website-v9 to lens design system, so every webpage under prisma (includes: blog, docs, homepage, website, etc) has the same header in order to keep consistency.

This will also hopefully handle the issues of the [navbar jumping](https://github.com/prisma/website-v9/issues/59), as the re-render of a slightly different component (implementation-wise) may be causing the visual flickering on pages that use the header from `lens` (i.e. /docs)

## Changes

- Removed `styled-components` package.
- Added `"@emotion/babel-preset-css-prop"` to storybook presets.
- Added the following packages:
  - "@emotion/core": "^11.0.0",
  - "@emotion/react": "^11.9.0",
  - "@emotion/styled": "^11.9.3"
- Added `<ThemeProvider />` to `LensWebProvider`. This should only influence the web components from lens.
- Deleted unused/old/deprecated files (i.e. `Dropdown`, `Navbar`, etc), in favor of new versions from `website-v9`, as well as replaced the `theme` value primitives and styling objects (for dark + default).

## UI/UX changes:
Screenshot 2022-07-28 at 10 24 25

- Button component update (to follow new design system- this should also be a reflected change on the Button component itself)
- Updates in paddings, alignment of active hover state (i.e. `Pricing` tab on hover shows a different styling than the dropdown counterparts)
Screenshot 2022-07-28 at 10 24 36

- Removal of unnecessary  `.svg` such as `chevron-up`, to use only `chevron-down` and transform it using CSS as needed (i.e. hover)
- Update of image used on the Header, for the blogpost, to use the optimal image format `.webp` instead of `.png`.
- Adding `->` arrow to identify external links.

https://user-images.githubusercontent.com/14851246/176143258-423714de-e02f-493c-b5f2-cef34183a860.mov

Added video from https://github.com/prisma/website-v9/pull/53 to demonstrate new UX behaviour to identify current page the user is at. This cannot be tested on storybook as it depends on the current location of the website.

| before | after |
| --- | --- |
|  Screenshot 2022-07-28 at 10 51 20 | Screenshot 2022-07-28 at 10 52 16 | 

## What issue does this fix?

[Fixes website-v9#62](https://github.com/prisma/website-v9/issues/62)

## Any other relevant information

`<ThemeProviders />` is seemingly not working, `theme` property is provided straight from the import, as it seems it is a recurring issue to use styled components together with storybook due to version incompatibilities. This is tackled by WebsiteHeader receiving a prop/attribute by the name of `lightTheme` that dictates the theme it falls to. Currently the default is `light`, so if any page requires a change for `dark`, it should do accordingly.

### CODE REVIEW

File: src/components/combo-box/ComboBox.tsx [hunk 1]
```diff
@@ -224,7 +222,7 @@ function ComboBoxContainer<OptionKey extends Key = string>({
   )
 }
 
-export const ComboBox = {
+export const ComboBox: any = {
```
Comment (tatianainama) [inline src/components/combo-box/ComboBox.tsx:225]: Same as before

**Review by tatianainama — CHANGES_REQUESTED** [review]
For this review, I focused mainly in changes that have an impact on PDP Control Plane

---


We are getting inferred types in every component as `Emotion.JSXElement`.
I think this is caused by the new configs introduced in `tsconfig.json` 

image

Declaring types fixes this

```jsx
const PickerContainer:FC<PickerProps> = ({ ... }) => { ... }
```

image

File: src/components/menu/Menu.tsx [hunk 2]
```diff
@@ -337,7 +337,7 @@ function MenuOption<Key extends string>({
   )
 }
 
-export const Menu = {
+export const Menu: any = {
```
Comment (tatianainama) [inline src/components/menu/Menu.tsx:340]: Same as https://github.com/prisma/lens/pull/410#discussion_r933044403

File: src/components/select/Select.tsx [hunk 3]
```diff
@@ -192,7 +192,7 @@ function SelectContainer<OptionKey extends Key = string>({
   )
 }
 
-export const Select = {
+export const Select: any = {
```
Comment (tatianainama) [inline src/components/select/Select.tsx:195]: Same as https://github.com/prisma/lens/pull/410#discussion_r933044403

File: src/components/combo-box/ComboBox.tsx [hunk 4]
```diff
@@ -93,7 +93,6 @@ function ComboBoxContainer<OptionKey extends Key = string>({
 
   const { contains } = useFilter({ sensitivity: "base" })
   const state = useComboBoxState({
-    id,
```
[the remark this answers is not in the document]
Comment (carlagn) [reply src/components/combo-box/ComboBox.tsx]: Having this variable results in this build error: 
```
[!] semantic error TS2345: Argument of type '{ id: string | undefined; autoFocus: boolean | undefined; children: CollectionChildren<ComboBoxOption<OptionKey>>; allowsEmptyCollection: true; ... 9 more ...; onSelectionChange: (k: Key) => void; }' is not assignable to parameter of type 'ComboBoxStateProps<ComboBoxOption<OptionKey>>'.
  Object literal may only specify known properties, and 'id' does not exist in type 'ComboBoxStateProps<ComboBoxOption<OptionKey>>'.
```

any clue on its origin? Reading the docs on react-stately I can't find any variable for `id` or similar, I might be missing something?

File: src/cloud/picker/Picker.tsx [hunk 5]
```diff
@@ -135,7 +135,7 @@ const PickerContainer = ({
   )
 }
 
-export const Picker = {
+export const Picker: any = {
```
[the remark this answers is not in the document]
Comment (carlagn) [reply src/cloud/picker/Picker.tsx]: The inference does not seem to be posing a problem for the build, so I'll revert it back

File: src/components/combo-box/ComboBox.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (tatianainama) [reply src/components/combo-box/ComboBox.tsx]: After some investigation, I realised this `id` shouldn't be passed here (as the error we are getting from TS hints 😓).
The `id` is later passed as its own prop to the `ListBoxOverlay` component, so we are good removing it, sorry for the confusion!

File: src/cloud/picker/Picker.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (tatianainama) [reply src/cloud/picker/Picker.tsx]: Uhm, I tested the new changes locally, and I'm still getting the following when building:

```
[!] (plugin rpt2) Error: lens/src/components/combo-box/ComboBox.tsx(225,14): semantic error TS2742: The inferred type of 'ComboBox' cannot be named without a reference to '@react-types/overlays/node_modules/@react-types/shared'. This is likely not portable. A type annotation is necessary.
```

Can we try again after applying the changes from https://github.com/prisma/lens/pull/410#discussion_r935418826 ?

[the remark this answers is not in the document]
Comment (carlagn) [reply src/cloud/picker/Picker.tsx]: [#410 (comment](https://github.com/prisma/lens/pull/410#discussion_r935418826) doesn't fix.

Build error is still the same as yours. Setting type as `Object` i.e.

```js
export const Picker: Object = {
```

works but might not be specific enough.

[the remark this answers is not in the document]
Comment (carlagn) [reply src/cloud/picker/Picker.tsx]: @tatianainama any thoughts on this fix? :/ ^

[the remark this answers is not in the document]
Comment (tatianainama) [reply src/cloud/picker/Picker.tsx]: After the upgrades, the error is gone 🎉

**Review by tatianainama — APPROVED** [review]
As I stated before, I focused mainly in changes that have an impact on PDP Control Plane, someone from Marketing team should take look at the changes this PR implies for the website.

Said that, everything looks good 😄

**Pull request merged**